### PR TITLE
[FIX] FilterEvaluationPlugin: insert/remove row on data filter header

### PR DESCRIPTION
**Pull request by dhrp-odoo**
## Description:

This PR includes two distinct commits that address issues related to the handling of data filter headers in our SheetPlugin and FilterEvaluationPlugin.

Commit 1: [FIX] FilterEvaluationPlugin: insert/remove row on data filter header

- Previously, when inserting a row above or below a data filter header, hidden rows inside the data filter were not being updated. This caused a state inconsistency issue between the data filter header and the data filter values.
- This commit address the problem by ensuring that hidden rows inside the data filter header are correctly updated when rows are inserted above or below the data filter header.

- Another issue occurred when removing a row above the data filter header while certain rows were filtered. This led to a mismatch in the data filter state, revealing hidden rows. 
- This commit fixes this problem by updating the handling of hidden rows inside the data filter header during row removal.

Commit 2: [FIX] SheetPlugin: removing row with data filter header

- Previously, removing rows that included the filter header failed to remove the associated data filter tables.
- This commit addresses the issue by ensuring the data filter tables is also removed when rows with the filter header are deleted.

Task: : [3546012](https://www.odoo.com/web#id=3546012&action=333&active_id=2328&model=project.task&view_type=form&cids=1&menu_id=4720)

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_lt("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

### CODE REVIEW

File: tests/plugins/filters.test.ts [hunk 1]
```diff
@@ -492,15 +492,9 @@ describe("Filters plugin", () => {
...
+      test("Removing rows with filter table should remove the filter table", () => {
         deleteRows(model, [2]);
-        expect(model.getters.getFilterTables(sheetId)[0].zone).toEqual(toZone("C3:F5"));
-        expect(getFilterValues(model)).toEqual([
-          { zone: "C3:C5", value: ["C"] },
-          { zone: "D3:D5", value: ["D"] },
-          { zone: "E3:E5", value: ["E"] },
-          { zone: "F3:F5", value: ["F"] },
-        ]);
+        expect(model.getters.getFilterTables(sheetId).length).toEqual(0);
```
Comment (rrahir) [inline tests/plugins/filters.test.ts:497]:
```suggestion
        expect(model.getters.getFilterTables(sheetId)).toEqual([]);
```

File: tests/plugins/filter_evaluation.test.ts [hunk 2]
```diff
@@ -164,4 +166,35 @@ describe("Filter Evaluation Plugin", () => {
...
+
+  test("Inserting rows above or below the data filter header updates the filtered rows", () => {
+    const model = new Model();
+
+    createFilter(model, "A1:A3");
+    setCellContent(model, "A2", "Hi");
+    setCellContent(model, "A3", "Hi");
+
+    updateFilter(model, "A1", ["Hi"]);
+    addRows(model, "before", 0, 1);
```
Comment (rrahir) [inline tests/plugins/filter_evaluation.test.ts:178]: for both tests ,check after/before and use the getter `isRowFiltered`, to ensure that it is the work of the filter itself.

**Review by rrahir — CHANGES_REQUESTED** [review]

**Comment by dhrp-odoo** [comment]
Hello @rrahir, I've implemented the suggested changes. Thank you!

**Comment by rrahir** [comment]
@robodoo r+

**Comment by rrahir** [comment]
@robodoo rebase-ff

**Pull request closed without merging**